test/prob/test_fixture_cdf_log.hpp

```diff
@@ -354,11 +354,15 @@ class AgradCdfLogTestFixture : public ::testing::Test {
         << "Number of finite diff gradients and calculated gradients must "
            "match -- error in test fixture";
     for (size_t i = 0; i < finite_dif.size(); i++) {
-      EXPECT_NEAR(finite_dif[i], gradients[i], 1e-4)
-          << "Comparison of finite diff to calculated gradient failed for i="
-          << i << ": " << parameters << std::endl
-          << "  finite diffs: " << finite_dif << std::endl
-          << "  grads:        " << gradients;
+      // use relative error check in first case as logs of functions do not
```

Comment:
I understand the reasoning for doing relative error checks away from zero and absolute error checks near zero. I'm not familiar with this log asymptotic thing though. Is it related?

Also I get our tolerances are bad here, but 10% off for the relative error seems high? Was that just an arbitrary choice or does it need to stay like that?

I'm not super concerned about these tests, but this changes touches other code yada yada so gotta be careful. I trust Bob's autodiff stuff and I'm pretty sure it's covering these tests (didn't realize this finite difference stuff was being checked already!). I'll ask Bob about this separately in this thread.

Reply:
The asymptotic thing is just from the fact that as a function asymptotically converges towards 0, the log of that function goes to -inf. When you've got a function that converges asymptotically then the absolue error is (hopefully) going to shrink the closer it gets to that value. For the other functions it's fine to use a single absolue error check over the whole function, but for log(x) functions you have to switch to a relative error check as x->0.

10% was a completely arbitrary choice. I've changed it ot 1% now (just checked and it's still passing all the tests).